receiver/stefreceiver/stef.go

```diff
@@ -0,0 +1,165 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package stefreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/stefreceiver"
+import (
+	"context"
+	"errors"
+	"net"
+	"sync"
+
+	stefgrpc "github.com/splunk/stef/go/grpc"
+	"github.com/splunk/stef/go/grpc/stef_proto"
+	"github.com/splunk/stef/go/otel/oteltef"
+	stefpdatametrics "github.com/splunk/stef/go/pdata/metrics"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentstatus"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/stefreceiver/internal"
+)
+
+type stefReceiver struct {
+	cfg        *Config
+	serverGRPC *grpc.Server
+
+	nextMetrics consumer.Metrics
+	shutdownWG  sync.WaitGroup
+
+	settings receiver.Settings
+}
+
+// Start runs the STEF gRPC receiver.
+func (r *stefReceiver) Start(_ context.Context, host component.Host) error {
+	var err error
+	if r.serverGRPC, err = r.cfg.ServerConfig.ToServer(
+		context.Background(), host, r.settings.TelemetrySettings,
+	); err != nil {
+		return err
+	}
+
+	r.serverGRPC = grpc.NewServer()
+
+	r.settings.Logger.Info("Starting GRPC server", zap.String("endpoint", r.cfg.ServerConfig.NetAddr.Endpoint))
+	var gln net.Listener
+	if gln, err = r.cfg.ServerConfig.NetAddr.Listen(context.Background()); err != nil {
+		return err
+	}
+
+	schema, err := oteltef.MetricsWireSchema()
+	if err != nil {
+		return err
+	}
+
+	settings := stefgrpc.ServerSettings{
+		Logger:       nil,
+		ServerSchema: &schema,
+		MaxDictBytes: 0,
+		Callbacks:    stefgrpc.Callbacks{OnStream: r.onStream},
+	}
+
+	stefSrv := stefgrpc.NewStreamServer(settings)
+	stef_proto.RegisterSTEFDestinationServer(r.serverGRPC, stefSrv)
+
+	r.shutdownWG.Add(1)
+	go func() {
+		defer r.shutdownWG.Done()
+		if errGrpc := r.serverGRPC.Serve(gln); errGrpc != nil && !errors.Is(errGrpc, grpc.ErrServerStopped) {
+			componentstatus.ReportStatus(host, componentstatus.NewFatalErrorEvent(errGrpc))
+		}
+	}()
```

Comment:
Could we use `errgroup.Group` instead of using a waitgroup?

Reply:
done